src/run/run.c

```diff
@@ -194,6 +198,7 @@ static int parse_argv(int argc, char *argv[]) {
                 { "unit",              required_argument, NULL, 'u'                   },
                 { "description",       required_argument, NULL, ARG_DESCRIPTION       },
                 { "slice",             required_argument, NULL, ARG_SLICE             },
+                { "slice-inherit",     optional_argument, NULL, ARG_SLICE_INHERIT     },
```

Comment:
hmm, getopt()'s `optional_argument` options are not nice to use, since they have this weird logic where they might or might not eat up the next argument. we are nowadays pretty careful with introducing new options that use this. e.g.:

```
systemd-run --slice-inherit foobar baz
```

if you look at the line above it could mean that you call the command the switch with no argument, and foobar and baz as separate comand line arguments. Or it could be the switch with foobar as argument, and only baz as separate command line argument. The glibc getopt code implements the second one, but it's not obvious to the reader I think.

Long story short. what about making `--slice-inherit` the long option unconditionally with argument, and then introuduce a short version `-I` or so that unconditionally takes no argument. so that you choose the version with/without argument by using the long/short option. 

Does that make sense?

Reply:
or, another idea: make --slice-inherit a switch without any arguments, and allow it to be combined with --slice=. If they are combined, then --slice= is the suffix appended to the caller's slice.

I think this would make a ton of sense, no?

Reply:
I have not a strong systemd-culture :) so whatever you think fits best. So '.' would mean current slice when combined with --slice-inherit otherwise it would mean the '..slice' ?